app/javascript/components/redux/app/app.js

```diff
@@ -0,0 +1,49 @@
+// --------- PATHS ------------------------------
+const UPLOAD_MESSAGES = 'REDUX/APP/APP/UPLOAD_MESSAGES';
+const FETCH_MESSAGES = 'REDUX/APP/APP/FETCH_MESSAGES';
+// -------------ACTIONS -----------------------
+const uploadMessages = (payload) => ({
+  type: UPLOAD_MESSAGES,
+  payload,
+});
+const fetchMessages = (payload) => ({
+  type: FETCH_MESSAGES,
+  payload,
+});
+
+// ----------- REDUCERS ----------------------
+
+const defaultValues = {
+  message: 'hello reducer',
+};
+
+const messagesReducer = (state = defaultValues, action) => {
+  switch (action.type) {
+    case UPLOAD_MESSAGES:
+      return action.payload;
+    default:
+      return state;
+  }
+};
+// ------------ MIDDLEWARES -------------------
+const fetchMessagesFromAPIMiddleware = (store) => (next) => (action) => {
+  if (action.type === FETCH_MESSAGES) {
+    fetch('/v1/messages', {
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    }).then((response) => response.json())
+      .then((json) => store.dispatch(uploadMessages(json)));
+  }
+  next(action);
+};
```

Comment:
[OPTIONAL]

- It's always better to follow modularity to make the codebase clean and not messy. It's better to keep separate files for different work.